alot/__main__.py

```diff
@@ -13,115 +14,67 @@
-    optParameters = [
-        ['sort', 'newest_first', None, 'Sort order'],
-    ]
-
-
-class Options(usage.Options):
-    optFlags = [["read-only", "r", 'open db in read only mode'], ]
-
-    def colourint(val):
-        val = int(val)
-        if val not in [1, 16, 256]:
-            raise ValueError("Not in range")
-        return val
-    colourint.coerceDoc = "Must be 1, 16 or 256"
-
-    def debuglogstring(val):
-        if val not in ['error', 'debug', 'info', 'warning']:
-            raise ValueError("Not in range")
-        return val
-    debuglogstring.coerceDoc = "Must be one of debug,info,warning or error"
-
-    optParameters = [
-        ['config', 'c', None, 'config file'],
-        ['notmuch-config', 'n', None, 'notmuch config'],
-        ['colour-mode', 'C', None, 'terminal colour mode', colourint],
-        ['mailindex-path', 'p', None, 'path to notmuch index'],
-        ['debug-level', 'd', 'info', 'debug log', debuglogstring],
-        ['logfile', 'l', '/dev/null', 'logfile'],
-    ]
-    search_help = "start in a search buffer using the querystring provided "\
-                  "as parameter. See the SEARCH SYNTAX section of notmuch(1)."
-
-    subCommands = [['search', None, SearchOptions, search_help],
-                   ['compose', None, ComposeOptions, "compose a new message"]]
-
-    def opt_version(self):
-        print alot.__version__
-        sys.exit(0)
-
 
 def main():
-    # interpret cml arguments
-    args = Options()
-    try:
-        args.parseOptions()  # When given no argument, parses sys.argv[1:]
-    except usage.UsageError as errortext:
-        print '%s' % errortext
-        print 'Try --help for usage details.'
-        sys.exit(1)
+    # set up the parser to parse the command line options.
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument('-v', '--version', action='version',
+                        version='%(prog)s, {}'.format(alot.__version__))
+    parser.add_argument('-r', '--read-only', action='store_true',
+                        help='open db in read only mode')
+    parser.add_argument('-c', '--config', type=argparse.FileType('r'),
+                        help='config file')
+    parser.add_argument('-n', '--notmuch-config', type=argparse.FileType('r'),
+                        help='notmuch config')
+    parser.add_argument('-C', '--colour-mode', choices=(1, 16, 256), type=int,
+                        help='terminal colour mode.')
+    parser.add_argument('-p', '--mailindex-path', #type=directory,
+                        help='path to notmuch index')
+    parser.add_argument('-d', '--debug-level', default='info',
+                        choices=('debug', 'info', 'warning', 'error'),
+                        help='debug log [default: info].')
+    parser.add_argument('-l', '--logfile', default='/dev/null',
+                        type=lambda x: argparse.FileType('w')(x).name,
```

Comment:
I used the argparse.FileType because it integrates nicely with usage messages if the user supplies bad files (not writeable/readable or so).

About your concerns: At least in the interactive python interpreter I get this result:
```python
>>> import argparse
>>> x = argparse.FileType('r')('somefile')
>>> # If I now `lsof -p $PID_OF_PYTHON` I see that somefile is opend
>>> del x
>>> # no lsof does not list it any longer
```
And as the FileType object is only used in the lambda to get the file name (as a string) I assume that the object is deleted afterwards. Is that right for lambdas? I know the lambda itself is kept in the parser object. But the object generated during its run should not be. The following test in the python interpreter seems to support this asumptions:
```python
>>> xx = lambda x: argparse.FileType('r')(x).name
>>> xx('somefile')
'somefile'
>>> # at this point lsof does not list somefile
>>> xx
<function <lambda> at 0x7f1596b51398>
```